frontend/__tests__/containers/ReportStep/DownloadDialog.test.tsx

```diff
@@ -0,0 +1,107 @@
+import { DateRangeItem, DownloadDialog } from '@src/containers/ReportStep/DownloadDialog';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+describe('DownloadDialog', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const handleCloseFn = jest.fn();
+  const downloadCSVFileFn = jest.fn();
+  const mockData = [
+    {
+      startDate: '2024-01-01',
+      endDate: '2024-01-14',
+      disabled: false,
+    },
+    {
+      startDate: '2024-01-15',
+      endDate: '2024-01-31',
+      disabled: false,
+    },
+  ];
+
+  const setup = (dateRangeList: DateRangeItem[]) => {
+    render(
+      <DownloadDialog
+        isShowDialog={true}
+        handleClose={handleCloseFn}
+        dateRangeList={dateRangeList}
+        downloadCSVFile={downloadCSVFileFn}
+      />,
+    );
+  };
+
+  it('should show all dateRanges in DownloadDialog', () => {
+    setup(mockData);
+
+    expect(screen.getByText('2024/01/01 - 2024/01/14')).toBeInTheDocument();
+    expect(screen.getByText('2024/01/15 - 2024/01/31')).toBeInTheDocument();
+    expect(screen.getAllByRole('checkbox')[0]).not.toBeDisabled();
+    expect(screen.getAllByRole('checkbox')[1]).not.toBeDisabled();
+    expect(screen.getByText('Confirm')).toBeDisabled();
+  });
+
+  it('should not be clickable when there is an error fetching data for this dataRange', () => {
```

Comment:
change `when` to `given` when necessary,
firstly, you need to define the `when` and `given` in test description

Reply:
Code has been changed.